samza-core/src/main/java/org/apache/samza/container/RunLoop.java

```diff
@@ -250,6 +285,40 @@ private void runTasks(IncomingMessageEnvelope envelope) {
     }
   }
 
+  /**
+   * when elasticity is not enabled, fetch the workers from sspToTaskWorkerMapping using envelope.getSSP()
+   * when elasticity is enabled,
+   *       sspToTaskWorkerMapping has workers for a SSP which has keyBucket
+   *       hence need to use envelop.getSSP(elasticityFactor)
+   *       Additionally, when envelope is EnofStream, it needs to be sent to all works for the ssp irrespective of keyBucket
```

Comment:
Could we list out all the message-types that map to more than one workes 
e.g., EOS, watermark, etc?